k-distribution/k-tutorial/2_intermediate/01_macros/README.md

```diff
@@ -4,6 +4,151 @@ copyright: Copyright (c) Runtime Verification, Inc. All Rights Reserved.
+                | "if" "(" Exp ")" Stmt "else" Stmt
+                | "{" Stmts "}"
+  syntax Stmts ::= List{Stmt,""}
+
+  rule if ( E ) S => if ( E ) S else { }
+endmodule
+```
+
+In this example, we see that an if statement without an `else` clause is
+defined in terms of one with an `else` clause. As a result, we would only
+need to give a single rule for how to rewrite `if` statements, rather than
+two separate rules for two types of `if` statements. This is a common pattern
+for dealing with program syntax that contains an optional component to it.
+
+It's worth noting, by default, macros are not applied recursively. To be more
+precise, by default a macro that arises as a result of the expansion of the
+same macro is not rewritten further. This is primarily to simplify the macro
+expansion process and reduce the risk that improperly defined macros will
+lead to non-terminating behavior.
+
+It is possible, however, to tell K to expand a macro recursively. To do this,
+simply replace the `macro` attribute with the `macro-rec` attribute. Note that
+K does not do any kind of checking to ensure termination here, so it is
+important that rules be defined correctly to always terminate, otherwise the
+macro expansion phase will run forever. In practice, the types of macros
+usually written are very easy to ensure this property on, however.
+
+### Exercise
+
+Using a `Nat` sort containing the constructors `0` and `S` (i.e., a
+peano-style axiomatization of the natural numbers where `S(N) = N + 1`,
+`S(S(N)) = N + 2`, etc), write a macro that will compute the sum of two
+numbers.
+
+## Aliases
+
+**NOTE**: future lessons will discuss KORE, an intermediate language for K,
+which is based on Matching Logic. KORE defines a concept called an "alias"
+which is similar, but not exactly the same as the concept we are about to
+introduce, which is a particular variant of a macro. It's worth remembering
+to keep these two concepts separate in your mind.
+
+Macros can be very useful in helping you define a programming language.
+However, they can be disruptive while pretty printing a configuration. For
+example, I might write a set of macros that transforms the code the user wrote
+into equivalent code that is slightly harder to read. This can make it more
+difficult to understand the code when it is pretty printed as part of the
+output of rewriting.
+
+K defines a relatively straightforward but novel solution to this problem,
+which is known as a K **alias**. An alias in K is very similar to a macro,
+with the exception that the rewrite rule will also be applied **backwards**
+during the pretty-printing process.
+
+It is very simple to make a production be an alias instead of a macro: simply
+use the `alias` or `alias-rec` attributes instead of the `macro` or `macro-rec`
+attributes. For example, if I were to declare the example involving `if`
+statements above using an alias instead of a macro, the `Stmt` term
```

Comment:
```suggestion
attributes. For example, if the example involving `if` statements
above was declared using an alias instead of a macro, the `Stmt` term
```